--- jboss-seam-2.0.2.SP1/src/main/org/jboss/seam/persistence/HibernatePersistenceProvider.java 
+++ jboss-seam-2.0.2.SP1_patched/src/main/org/jboss/seam/persistence/HibernatePersistenceProvider.java 
@@ -4,6 +4,7 @@
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
+import java.util.Collection;
 import java.util.Map;
 
 import javax.persistence.EntityManager;
@@ -203,7 +204,12 @@
          org.hibernate.Filter filter = getSession(entityManager).enableFilter( f.getName() );
          for ( Map.Entry<String, ValueExpression> me: f.getParameters().entrySet() )
          {
-            filter.setParameter( me.getKey(), me.getValue().getValue() );
+		   Object filterValue = me.getValue().getValue();
+		   if ( filterValue instanceof Collection ) {
+		      filter.setParameterList(me.getKey(), (Collection) filterValue);
+		   } else {
+			  filter.setParameter(me.getKey(), filterValue);
+		   }
          }
          filter.validate();
       }
