--- jboss-seam-2.0.2.SP1/src/main/org/jboss/seam/persistence/ManagedHibernateSession.java 
+++ jboss-seam-2.0.2.SP1_patched/src/main/org/jboss/seam/persistence/ManagedHibernateSession.java 
@@ -3,6 +3,7 @@
 
 import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
@@ -101,7 +102,12 @@
       org.hibernate.Filter filter = session.enableFilter( f.getName() );
       for ( Map.Entry<String, ValueExpression> me: f.getParameters().entrySet() )
       {
-         filter.setParameter( me.getKey(), me.getValue().getValue() );
+	     Object filterValue = me.getValue().getValue();
+		 if ( filterValue instanceof Collection ) {
+		    filter.setParameterList(me.getKey(), (Collection) filterValue);
+		 } else {
+			filter.setParameter(me.getKey(), filterValue);
+		}
       }
       filter.validate();
    }
